### ISSUE TITLE
[SSR] Custom directives on the server bundle

### ISSUE THREAD

**Issue by wucdbm** [vuejs/docs#1122]
I don't seem to be able to find any documentation on how to handle custom directives in a vue/vue.config.js/webpack setup using latest 3.1.* Vue. I have resorted to not using directives, but these would be a handy tool in our toolbox.

Does anyone know how to achieve this, or where I could find documentation on the matter?
Or is this still undocumented and needs to be added to the documentation pages?

**Comment by skirtles-code** [comment]
There isn't currently any documentation apart from a suggestion to avoid custom directives.

From a bit of digging in the source code I got something working. **This is not a recommendation. It is just something I came up with from browsing through the code.**

```js
// options to pass to vue-loader
compilerOptions: {
  directiveTransforms: {
    abcd (dir, node, context) {
      return {
        props: [],
        ssrTagParts: [' data-example="value"']
      }
    }
  }
}
```

This is for a `v-abcd` directive that adds a `data-example="value"` attribute to the element.

For Vue CLI it'd need `chainWebpack`. There's an example of using `compilerOptions` here: https://v3.vuejs.org/guide/migration/migration-build.html#installation
